src/connection.cpp

```diff
@@ -49,20 +49,18 @@ void connection::handle_read(const boost::system::error_code& e,
     boost::tie(result, boost::tuples::ignore) = request_parser_.parse(request_, buffer_.data(), buffer_.data() + bytes_transferred);
 
     if (result) {
-
-      // TODO: request type isn't being set correctly, need to set up logging to debug
-
-      // Use parsed request to determine whether it wants an echo or a file
-      // if (request_.request_type == request::static_file) {
-      //   request_handler_static_.handle_request(request_, reply_);
-      // }
-      // else {
-      //   request_handler_echo_.handle_request(request_, reply_);
-      // }
-      request_handler_echo_.handle_request(request_, reply_);
-      boost::asio::async_write(socket_, reply_.to_buffers(),
-        boost::bind(&connection::handle_write, shared_from_this(),
-          boost::asio::placeholders::error));
+      if (request_.uri == "/echo"){
+        request_handler_echo_.handle_request(request_, reply_);
+        boost::asio::async_write(socket_, reply_.to_buffers(),
```

Comment:
I think we can send the `boost::asio` stuff to under this if statement to avoid duplicating code